packages/plugins/i18n/shared/contracts/content-manager.ts

```diff
@@ -0,0 +1,29 @@
+import { Entity } from '@strapi/types';
+import { errors } from '@strapi/utils';
+
+/**
+ * POST /i18n/content-manager/actions/actions/get-non-localized-fields - Get the localized fields
+ */
+export declare namespace GetNonLocalizedFields {
+  export interface Request {
+    query: {};
+    body: {
+      id?: Entity.ID;
+      locale?: string;
+      model: string;
+    };
+  }
+
+  /**
+   * TODO: this should follow the usual `data/error` pattern.
+   */
+  export type Response =
+    | {
+        nonLocalizedFields: object;
+        localizations: Array<{ id: Entity.ID; locale: string; publishedAt: string | null }>;
+      }
+    | {
+        data: null;
+        error: errors.ApplicationError;
+      };
+}
```

Comment:
In https://github.com/strapi/strapi/pull/18918#discussion_r1409618327 Mark mentioned this doesn't play well with rtk-query.
What is normally more useful UI wise to have? Imo the format you propose is technically more accurate

Reply:
I've changed this one, the other issue is that other endpoints don't return an object, so it's not actually possible – i think for D&P we can refactor these and put them in our "typical" format which would help everything